### PR TITLE
C, Pylibcugraph, and Python API Updates for Edge Types

### LINKED ISSUE #2576: Edge Property Support - Create pylibcugraph interface for defining edge properties
Update pylibcugraph interface for graph creation to support edge id, edge type and weight.

More extensive property support will be considered later.

### LINKED ISSUE #2577: Edge Property Support - Create C API
Add support for both edge id, edge type id and weight in C API for graph creation.

Support for more complex edge properties will be addressed later.

### LINKED ISSUE #2579: Edge Property Support - Python
Provide support in python for creating an edge with both edge id and weight.

Blocked by https://github.com/rapidsai/cugraph/issues/2576 to start work.
Blocked by https://github.com/rapidsai/cugraph/issues/2575 to finish work.

### PR DESCRIPTION
**Pull request by alexbarghi-nv**
- [x] C API Changes
- [x] Pylibcugraph API Changes

Closes #2577 
Closes #2576 
Closes #2579 

<s>Note: this adds a UINT8 type which is the preferred data type for edge types.  The upcoming updates to the C++ code should only specialize `uint8_t` for `edge_type_t` in order to avoid unnecessary compilation.</s>

- [x] Update to work with `int32_t`

### CODE REVIEW

**Comment by ChuckHastings** [comment]
rerun tests

**Review by rlratzel — COMMENTED** [review]
Looks good, thanks for tackling this one.  I just had a very minor suggestion:

File: python/cugraph/cugraph/structure/graph_implementation/simpleDistributedGraph.py [hunk 1]
```diff
@@ -91,7 +106,9 @@ def _make_plc_graph(
             weight_array=values,
             store_transposed=store_transposed,
             num_edges=num_edges,
-            do_expensive_check=False
+            do_expensive_check=False,
+            edge_id_array=values_id,
+            edge_type_array=values_etype
```
[the remark this answers is not in the document]
Comment (alexbarghi-nv) [reply python/cugraph/cugraph/structure/graph_implementation/simpleDistributedGraph.py]: Just changed it.

**Comment by alexbarghi-nv** [comment]
rerun tests

**Review by rlratzel — APPROVED** [review]

**Comment by alexbarghi-nv** [comment]
rerun tests

**Comment by alexbarghi-nv** [comment]
rerun tests

**Comment by BradReesWork** [comment]
rerun tests

**Comment by BradReesWork** [comment]
@gpucibot merge

**Pull request merged**